rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java

```diff
@@ -495,6 +496,83 @@ public void should_document_create_project() throws Exception {
                         )));
     }
 
+    @Test
+    public void should_document_update_project() throws Exception {
+        Project updateProject = new Project();
+        updateProject.setName("updated project");
+        updateProject.setDescription("Project description updated");
+        updateProject.setVersion("1.0");
+        String accessToken = TestHelper.getAccessToken(mockMvc, testUserId, testUserPassword);
+        this.mockMvc
+                .perform(patch("/api/projects/376576").contentType(MediaTypes.HAL_JSON)
+                        .content(this.objectMapper.writeValueAsString(updateProject))
+                        .header("Authorization", "Bearer " + accessToken).accept(MediaTypes.HAL_JSON))
+                .andExpect(status().isOk())
+                .andDo(this.documentationHandler.document(
+                links(linkWithRel("self").description("The <<resources-projects,Projects resource>>")),
+                requestFields(
+                        fieldWithPath("name").description("The name of the project"),
+                        fieldWithPath("type").description("project"),
+                        fieldWithPath("version").description("The version of the new project"),
+                        fieldWithPath("visibility").description("The project visibility, possible values are: "
+                                + Arrays.asList(Visibility.values())),
+                        fieldWithPath("description").description("The project description"),
+                        fieldWithPath("projectType").description("The project type, possible values are: "
+                                + Arrays.asList(ProjectType.values())),
+                        fieldWithPath("securityResponsibles").description("An array of users responsible for security of the project.")),
+                responseFields(fieldWithPath("name").description("The name of the project"),
+                        fieldWithPath("version").description("The project version"),
+                        fieldWithPath("createdOn").description("The date the project was created"),
```

Comment:
I just stumbled upon that field because I think from the UI, the user is not able to modify the `createdOn` field. There might be others as well. Do we want to let users change it via REST?

Reply:
Ah, I just checked that `org.eclipse.sw360.datahandler.db.ProjectDatabaseHandler.copyImmutableFields(Project, Project)` will not let any service user change the fields. So the behavior might be correct, but the documentation is misleading.

Reply:
Oh man, sorry for this monologue, but the documentation is alright. I just did not see in this diff view that there has been already the call of `responseFields()` before.
Just one question: What happens if one just passes another valid field of a project inside the request (one of the undocumented)? Will it be updated? Or is there some logic somewhere that restricts the patch request to the fields that you documented here?

Reply:
The same update behavior is followed for project as we have for component and release. Any editable field can be provided inside request json to be updated. I gave few fields as part of sample request json.

Reply:
I have updated the rest documentation accordingly.